src/ApplePayGateway.php

```diff
@@ -0,0 +1,206 @@
+ *    //If successful, the payment sheet should be fully loaded.
+ *    $apple_pay_session = $authorizeResponse->getPaymentSessionObject();
+ *
+ *     // TODO: Add functionality for completeAuthorize() and capture().
+ *     // An opaque Apple Pay Session is returned as a response (expires after 5 mins) and it can be sent to
+ *     // the front end to fully load the payment sheet. This allows the user to optionally configure their
+ *     // payment option and shipping methods (if needed) and submit their payment.
+ * </code>
+ */
+class ApplePayGateway extends AbstractVindiciaGateway
+{
+
+    /**
+     * Get the gateway name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return 'Vindicia ApplePay';
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getPemCertPath()
+    {
+        return $this->getParameter('pemCertPath');
+    }
+
+    /**
+     * A certificate associated with your merchant ID.
+     * Example:
+     * setPemCertPath('./certs/path_to_cert.crt.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setPemCertPath($value)
+    {
+        return $this->setParameter('pemCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPath()
+    {
+        return $this->getParameter('keyCertPath');
+    }
+
+    /**
+     * A key certificate associated with your merchant ID.
+     * Example:
+     * setKeyCertPath('./certs/path_to_cert.key.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setKeyCertPath($value)
+    {
+        return $this->setParameter('keyCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPassword()
+    {
+        return $this->getParameter('keyCertPassword');
+    }
+
+    /**
+     * Password associated with your key certificate.
+     * Can be omitted if a password isn't set.
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setKeyCertPassword($value)
+    {
+        return $this->setParameter('keyCertPassword', $value);
+    }
+
+    /**
+     * Makes request to Apple to set up session between Vimeo and Apple.
+     *
+     * @param array $parameters
+     * @return \Omnipay\Vindicia\Message\ApplePayAuthorizeRequest
+     */
+    public function authorize(array $parameters = array())
+    {
+        /**
+         * @var \Omnipay\Vindicia\Message\ApplePayAuthorizeRequest
+         */
+        return $this->createRequest('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest', $parameters);
+    }
+
+    /**
+     * Authorize an Apple Pay purchase.
+     *
+     * @param array $parameters
+     * @return \Omnipay\Common\Message\AbstractRequest
+     */
+    public function completeAuthorize(array $parameters = array())
```

Comment:
Same issue here about all possible valid $parameter values.

Reply:
I added a TODO in the docblock to add documentation and functionality for completeAuthorize() and capture() later.